From 548b53b17449f08d7e67bdd782de635125f30bc3 Mon Sep 17 00:00:00 2001
Message-Id: <548b53b17449f08d7e67bdd782de635125f30bc3.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Tue, 20 Jan 2015 17:51:47 +0530
Subject: [PATCH 09/29] MUST_REBASE [VPG]: video/adf: Get the platform id from
 VBT

VBT contains the gen platform id and I915 driver parses that during
the init time. This patch uses the adf wrapper function to get the
platform gen id.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

For: GMINL-5194
Change-Id: I623eaeb5972466827d95ef6027d49f3d0c3fcdce
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/video/adf/intel/core/intel_dc_config.c |    7 ++++++-
 drivers/video/adf/intel/intel_adf.c            |    6 +++---
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/drivers/video/adf/intel/core/intel_dc_config.c b/drivers/video/adf/intel/core/intel_dc_config.c
index e7d7ba8..af0c96a 100644
--- a/drivers/video/adf/intel/core/intel_dc_config.c
+++ b/drivers/video/adf/intel/core/intel_dc_config.c
@@ -17,7 +17,12 @@
 
 static const struct intel_dc_config_entry g_dc_configs[] = {
 	{
-		.id = 0,
+		.id = gen_valleyview,
+		.get_dc_config = vlv_get_dc_config,
+		.destroy_dc_config = vlv_dc_config_destroy,
+	},
+	{
+		.id = gen_cherryview,
 		.get_dc_config = vlv_get_dc_config,
 		.destroy_dc_config = vlv_dc_config_destroy,
 	},
diff --git a/drivers/video/adf/intel/intel_adf.c b/drivers/video/adf/intel/intel_adf.c
index 2b8de73..c9fcf68 100644
--- a/drivers/video/adf/intel/intel_adf.c
+++ b/drivers/video/adf/intel/intel_adf.c
@@ -11,7 +11,7 @@
  * GNU General Public License for more details.
  *
  */
-
+#include <drm/i915_adf.h>
 #include <intel_adf.h>
 
 static const struct intel_adf_context *g_adf_context;
@@ -215,10 +215,10 @@ struct intel_adf_context *intel_adf_context_create(struct pci_dev *pdev)
 #if defined(CONFIG_ADF_FBDEV) && defined(CONFIG_ADF_INTEL_FBDEV)
 	struct adf_fbdev *fbdevs;
 #endif
-	/*TODO: use real platform ID*/
-	u32 platform_id = 0;
+	u32 platform_id;
 	int err;
 
+	platform_id = (u32) intel_adf_get_platform_id();
 	if (!pdev)
 		return ERR_PTR(-EINVAL);
 
-- 
1.7.9.5

